plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestTimestampCoercer.java

```diff
@@ -102,6 +120,46 @@ public void testHistoricalLongTimestampToVarchar()
                 .hasMessageContaining("Coercion on historical dates is not supported");
     }
 
+    @Test(dataProvider = "invalidValue")
+    public void testInvalidVarcharToShortTimestamp(String invalidValue)
+    {
+        assertVarcharToShortTimestampCoercions(createUnboundedVarcharType(), Slices.utf8Slice(invalidValue), TIMESTAMP_MICROS, null);
+    }
+
+    @Test(dataProvider = "invalidValue")
+    public void testInvalidVarcharLongTimestamp(String invalidValue)
+    {
+        assertVarcharToLongTimestampCoercions(createUnboundedVarcharType(), Slices.utf8Slice(invalidValue), TIMESTAMP_MICROS, null);
+    }
+
+    @Test
+    public void testHistoricalVarcharToShortTimestamp()
+    {
+        LocalDateTime localDateTime = LocalDateTime.parse("1899-12-31T23:59:59.999999");
+        SqlTimestamp timestamp = SqlTimestamp.fromSeconds(TIMESTAMP_MICROS.getPrecision(), localDateTime.toEpochSecond(UTC), localDateTime.get(NANO_OF_SECOND));
+        assertThatThrownBy(() -> assertVarcharToShortTimestampCoercions(
+                createUnboundedVarcharType(),
+                Slices.utf8Slice("1899-12-31 23:59:59.999999"),
+                TIMESTAMP_MICROS,
+                timestamp.getEpochMicros()))
+                .isInstanceOf(TrinoException.class)
+                .hasMessageContaining("Coercion on historical dates is not supported");
```

Comment:
Maybe we can reformat this

```suggestion
        assertThatThrownBy(() ->
                assertVarcharToShortTimestampCoercions(
                        createUnboundedVarcharType(), 
                        Slices.utf8Slice("1899-12-31 23:59:59.999999"), 
                        TIMESTAMP_MICROS, 
                        timestamp.getEpochMicros()))
                .isInstanceOf(TrinoException.class)
                .hasMessageContaining("Coercion on historical dates is not supported");
```